src/backend/distributed/executor/local_executor.c

```diff
@@ -382,13 +382,12 @@ ExecuteLocalTaskListExtended(List *taskList,
 
 /*
  * SetColocationIdAndPartitionKeyValueForTasks sets colocationId and partitionKeyValue
- * for the tasks in the taskList if workerJob has a colocationId and partitionKeyValue.
+ * for the tasks in the taskList.
  */
 static void
 SetColocationIdAndPartitionKeyValueForTasks(List *taskList, Job *workerJob)
 {
-	if (workerJob->colocationId != 0 &&
-		workerJob->partitionKeyValue != NULL)
+	if (workerJob->colocationId != 0)
```

Comment:
nit: For readability let's use `INVALID_COLOCATION_ID`:

```suggestion
	if (workerJob->colocationId != INVALID_COLOCATION_ID)
```

I noticed we were comparing to this named constant in other places.